internal/credentials/kubernetes/github/app.go

```diff
@@ -198,13 +268,14 @@ func tokenCacheKey(
 	clientID string,
 	installationID int64,
 	encodedPrivateKey string,
+	allowedRepos []string,
 ) string {
 	return fmt.Sprintf(
 		"%x",
 		sha256.Sum256([]byte(
 			fmt.Sprintf(
-				"%s:%s:%d:%s",
-				baseURL, clientID, installationID, encodedPrivateKey,
+				"%s:%s:%d:%s:%s",
+				baseURL, clientID, installationID, encodedPrivateKey, strings.Join(allowedRepos, ","),
```

Comment:
See https://github.com/akuity/kargo/pull/4995/files#r2328386620

If we scope all installation tokens to just the one repo we're trying to access, then the cache key needs to include that repoURL, which means, baseURL doesn't need to be included anymore.

```suggestion
				repoURL, clientID, installationID, encodedPrivateKey, strings.Join(allowedRepos, ","),
```

You have to make corresponding adjustments to other bits of code to make `repoURL` available here _in place of_ both `baseURL` and `allowedRepos`.